src/ImageSharp/Formats/Icon/IconDecoderCore.cs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using SixLabors.ImageSharp.Formats.Bmp;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.ImageSharp.IO;
+using SixLabors.ImageSharp.Metadata;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats.Icon;
+
+internal abstract class IconDecoderCore(DecoderOptions options) : IImageDecoderInternals
+{
+    private IconDir fileHeader;
+
+    public DecoderOptions Options { get; } = options;
+
+    public Size Dimensions { get; private set; }
+
+    protected IconDir FileHeader { get => this.fileHeader; private set => this.fileHeader = value; }
+
+    protected IconDirEntry[] Entries { get; private set; } = [];
```

Comment:
SA1010
Should I avoid this syntax?

Reply:
You should be fine now that I've updated the StyleCop version.